scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -902,19 +902,24 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
       val startPos = auto.startTokenPos
       val res = token match {
         case LeftParen() => makeTupleType(startPos, inParensOnOpen(types()))
-        case Underscore() => next(); Type.Placeholder(typeBounds())
         case MacroSplicedIdent(ident) =>
           Type.Macro(macroSplicedIdent(ident))
         case MacroSplice() =>
           Type.Macro(macroSplice())
+        case _: Underscore =>
+          next()
+          if (dialect.allowUnderscoreAsTypePlaceholder)
+            Type.Placeholder("")
```

Comment:
```suggestion
            Type.Placeholder("_")
```
?

Reply:
the parameter doesn't refer to the wildcard but the variant prefix (covariant or contravariant).